### PR TITLE
Separate projectid/page param validation

### PR DESCRIPTION
**Pull request by cpeel**
Create two new projectID and page name param fetching functions that
more closely mirror the syntax for the other param fetching functions
in misc.inc. This lets us make the validate_*() functions purely
for validation and more generally useful elsewhere.

### CODE REVIEW

File: pinc/Project.inc [hunk 1]
```diff
@@ -1068,37 +1069,80 @@ function does_project_page_table_exist($projectid)
...
+    }
+    return $projectid;
 }
 
-// Checks if $value contains only legitimate characters for
-// the filename of a page image file in a project.
-// If it does, it returns it, otherwise dies with a warning.
-function validate_page_image_filename($param_name, $value, $allownull = false)
+// Confirm that $page_name is a well-formed page filename or raise an exception.
+function validate_page_image($page_name)
```
Comment (chrismiceli) [inline pinc/Project.inc:1111]: do we have unit tests for this? It seems simple enough to write one.

Comment (cpeel) [reply pinc/Project.inc:1111]: No we don't but we should - I will fix that.

Comment (cpeel) [reply pinc/Project.inc:1111]: Tests added.

Comment (chrismiceli) [reply pinc/Project.inc:1111]: Looks great thanks!

**Comment by cpeel** [comment]
Note to self: update the noncvs code when this gets rolled out to TEST and then PROD.

**Pull request merged**